src/GoTrueAdminApi.ts

```diff
@@ -156,21 +163,47 @@ export default class GoTrueAdminApi {
    * Get a list of users.
    *
    * This function should only be called on a server. Never expose your `service_role` key in the browser.
-   * @param params An object which supports `page` and `per_page` as numbers, to alter the paginated results.
+   * @param params An object which supports `page` and `perPage` as numbers, to alter the paginated results.
    */
   async listUsers(
     params?: PageParams
-  ): Promise<{ data: { users: User[] }; error: null } | { data: { users: [] }; error: AuthError }> {
+  ): Promise<
+    | { data: { users: User[]; aud: string } & Pagination; error: null }
+    | { data: { users: [] }; error: AuthError }
+  > {
     try {
-      const { data, error } = await _request(this.fetch, 'GET', `${this.url}/admin/users`, {
+      let pagination: Pagination = {}
+      const response = await _request(this.fetch, 'GET', `${this.url}/admin/users`, {
         headers: this.headers,
+        noResolveJson: true,
         query: {
           page: params?.page?.toString() ?? '',
           per_page: params?.perPage?.toString() ?? '',
         },
+        xform: _noResolveJsonResponse,
       })
-      if (error) throw error
-      return { data: { ...data }, error: null }
+      if (response.error) throw response.error
+
+      const links = response.headers.get('link')?.split(',')
+
+      if (links) {
+        links.forEach((link: string) => {
+          const url = link.split(';')[0].replace(/[\<\>\s]/g, '')
+          const rel = JSON.parse(
+            link
+              .split(';')[1]
+              .replace(/[\<\>\s]/g, '')
+              .split('=')[1]
+          )
+          pagination[rel] = url
+        })
+
+        if (!pagination.next) pagination.next = null
+        const total = response.headers.get('x-total-count') ?? '0'
+        pagination.total = parseInt(total)
+      }
+      const users = await response.json()
+      return { data: { ...users, ...pagination }, error: null }
```

Comment:
hmm the response returned here is something like the following:

```json
{ 
  users: [ ... ], 
  aud: 'authenticated',
  next: '/admin/users?page=3&per_page=3',
  last: '/admin/users?page=34&per_page=3',
  total: 100
}
```

i'm not too sure how useful the `next` and `last` properties are, maybe it will be more useful to rename this to `nextPage` and `lastPage` and just return the number instead of the url path. Wdyt?

Reply:
i think we don't have to return the `per_page` value, we can just return the page numbers ?

Reply:
oops, might have missed that - thanks for being so responsive thus far though!